src/foam/nanos/crunch/ui/CapabilityView.js

```diff
@@ -0,0 +1,105 @@
+foam.CLASS({
+  package: 'foam.nanos.crunch.ui',
+  name: 'MyView',
+  extends: 'foam.u2.View',
+  documentation: 'All purpose capability view that takes in an array of capaiblity ids and display these capabilities',
+
+  imports: [
+    'crunchController',
+    'notify'
+  ],
+
+  requires: [
+    'foam.u2.detail.SectionView'
+  ],
+
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'capabilities'
+    },
+    {
+      name: 'wizardlets',
+      documentation: 'wizardlets for the capability'
+    },
+    {
+      name: 'wizardletSectionsList',
+      documentation: `
+        sections for wizardlets
+        wizardletSectionsList[i] stores sections for wizardlets[i]
+      `
+    }
+  ],
+
+  methods: [
+    async function initE() {
+      const self = this;
+
+      this.start().addClass(this.myClass())
+        .add(this.slot(function(capabilities) {
+          return this.E().forEach(capabilities, async function(cap) {
+            // get wizardlets for the capability and all of its dependent capabilities
+            self.wizardlets = (await self.crunchController.getCapsAndWizardlets(cap)).wizCaps;
+
+            // add listeners on wizardlets
+            for ( const wizardlet of self.wizardlets ) {
+              self.addListeners(wizardlet.data);
+            }
+
+            // get all the sections associated with the wizardlets
+            self.wizardletSectionsList = self.crunchController.generateSections(self.wizardlets);
+            
+            this.add(self.slot(function(wizardletSectionsList) {
+              return this.E().forEach(wizardletSectionsList, function(sections, index) {
+                sections.map(section => (
+                  this.tag(self.SectionView, {
+                    section,
+                    data: self.wizardlets[index].data,
+                    showTitle: false
+                  })
+                ));
+              });
+            }));
+          });
+        }))
+      .end();
+    },
+
+    function addListeners(obj) {
+      if ( ! this.isFObject(obj) ) return;
+
+      obj.sub(this.updateWizardlet);
+
+      // add listeners on inner fobjects
+      for ( const value of Object.values(obj.instance_) ) {
+        if ( this.isFObject(value) ) {
+          this.addListeners(value);
+        }
+      }
+      return;
+    },
+
+    function isFObject(obj) {
+      if ( typeof obj === 'object' && obj.instance_ ) {
+        return true;
+      }
+    }
+  ],
+
+  listeners: [
+    {
+      name: 'updateWizardlet',
+      code: function() {
+        // TODO: can we only update the wizardlet that has new data instead of updating all the wizardlets?
+        // TODO: when/where do we call finalOnClose?
```

Comment:
we need to make some mods to finalOnClose - and will keep you posted. However I'm under the impression that it can be called here, after save.

Reply:
Don't we want to call the function at the end of the process (thus the name)? It looks an overkill to call the function each time listeners are hit.

Reply:
I agree it does seem messy, however I would argue that this should be a minimal call due to the nature of this view. 
1) This view is not for a full feature, it is for a few capabilities that are dependencies of legacy objects. 
2) This view is also for getting additional requirements from within a feature ( example: our invoice flow concept with paymentProvidorCapabilities), however again I would argue that these capabilities will be for additional information, simple dynamic capabilities that will have little more than a few properties. Therefore the call to the listener will be minimal.